src/components/OptionsList/BaseOptionsList.js

```diff
@@ -121,8 +120,12 @@ class BaseOptionsList extends Component {
 
             // Add section items
             for (let i = 0; i < section.data.length; i++) {
-                flatArray.push({length: optionHeight, offset});
-                offset += optionHeight;
+                let fullOptionHeight = variables.optionRowHeight;
+                if (i > 0 && this.props.shouldHaveOptionSeparator) {
+                    fullOptionHeight += variables.borderTopWidth;
+                }
+                flatArray.push({length: fullOptionHeight, offset});
+                offset += fullOptionHeight;
```

Comment:
@rushatgabhane I think the above might be incorrect since we're calculating the offset is supposed to be for **this item**. Meaning we should not add this item's height to the offset when pushing to the `flatArray`. That's why the offset calculation happens after the `flatArray.push` in the current code, the **next item** will use this calculated offset, not this item

```
flatArray.push({length: fullOptionHeight, offset});
offset += fullOptionHeight;
```